### PR TITLE
Temp feature

### PR DESCRIPTION
**Pull request by aryangoel24**
final mongodb implementation done with the help of aviral
added empty constructors to User and Order class
added get methods for each User attribute
changed all dateOrdered instances to string versions of LocalDateTime

### CODE REVIEW

File: src/main/java/mainMongoDB.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+//        System.out.println(findAttributeByUsername("nikita", "pastOrders"));
+
+//        System.out.println(findPastOrders("ag24"));
+//        System.out.println(findPastOrders("ag24").getLastOrdered());
+//        System.out.println(findPastOrders("ag24").pastOrdersMap.get("2022-11-16T15:09:09.288516"));
+//        System.out.println(userRepo.find(eq("username", "rg66")).first().getPastOrders().getOrderedItemsByDate("2022-11-16T17:53:27.399046"));
+//        Order order = new Order(LocalDateTime.now().toString());
+//        foodItem f1 = new foodItem("Hamburger", 20);
+//        order.addToOrder(f1);
+//        addToPastOrders("aryangoel24", order);
```
Comment (Aviral-03) [inline src/main/java/mainMongoDB.java:131]: Delete the redundant code.

File: src/main/java/mainMongoDB.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+//        return user.getBudget();
+//    }
+
+    public static void updateBudget (String username) {
+
+    }
+
+//    Update basic User or Restaurant attribute
+
+    public static void updateAttributeByUsername (String username, String attribute, String attValue) {
```
Comment (Aviral-03) [inline src/main/java/mainMongoDB.java:217]: Add another method for restaurant method which will extract attributes from the given input

**Review by Aviral-03 — CHANGES_REQUESTED** [review]
Overall good job, you have identified every aspect for the database, however there are several changes to be made. I have added two comments in the file, where some of the redundant code is written. Furthermore, Clean Architecture is little absent because the gateway and requestModel for the database is not created. However, you can add that later. But review the changes that I have commented in your file. Additionally, you could also create tests for MongoDB file.

File: src/main/java/mainMongoDB.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        pastOrders.addOrder(order);
+
+        Document query = new Document("username", username);
+        Bson updates = Updates.set("pastOrders", pastOrders);
+        UpdateOptions options = new UpdateOptions().upsert(true);
+
+        userRepo.updateOne(query, updates, options);
+        return true;
+    }
+
```
Comment (Aviral-03) [inline src/main/java/mainMongoDB.java:101]: Add another method here to extract the menu item from the restaurant collection from MongoDB

**Review by Aviral-03 — CHANGES_REQUESTED** [review]

**Review by Aviral-03 — DISMISSED** [review]
Everything looks good to me. You can add tests for this class and work on the clean architecture connecting this class to gateways and use case interactor, then create another pull request so that that can be merged.

File: src/main/java/Entities/Order.java [hunk 4]
```diff
@@ -5,29 +5,21 @@
 import java.util.ArrayList;
 
 public class Order {
-    protected ArrayList<foodItem> orderedItems;
-    protected LocalDateTime dateOrdered;
-    protected double totalCost;
-    protected Restaurant restaurant;
```
Comment (nathan-cai) [inline src/main/java/Entities/Order.java:11]: Why does order no longer keep track of which restaurant it is for?

**Review by nathan-cai — CHANGES_REQUESTED** [review]
The database setup looks good. However, I noticed that you guys removed an instance variable fo restaurant in the order entity, I think we need this because it is important to show the user where the order came form in past orders. Creating gateways is the next step so that the rest of our classes can interact with the database. Good Job!

File: src/main/java/Entities/Order.java (hunk 4 above)
Comment (aryangoel24) [reply src/main/java/Entities/Order.java:11]: This is an important design decision, and based on group discussion, I have added it back and adjusted the relevant classes to account for this instance attribute.

**Review by nathan-cai — APPROVED** [review]

**Review by darpanmishra — APPROVED** [review]
Looks good to me.
Great job implementing our Database with Mongo.
Let’s see how we can connect our use cases to our presenters.

**Pull request merged**